### ISSUE TITLE
Use dropdowns rather than radio buttons

### ISSUE THREAD

**Issue by timquinlan** [coder/coder#10264]
User feedback: Can we add something to the coder_parameter options to display the options as a dropdown menu rather than radio buttons?  This would save screen space when creating workspaces from our most complex templates.

**Comment by kylecarbs** [comment]
Agreed!

**Comment by matifali** [comment]
I think we should do it without changing anything in the provider and should convert any parameter with more than 4 options to a drop-down

**Comment by timquinlan** [comment]
@matifali I think there should be an option in the provider to allow the user to choose.

**Comment by matifali** [comment]
@timquinlan I agree. so it can probably work like this
```hcl
data "coder_paramter" "options" {
	name         = "options"
	display_name = "Options to choose"
	description  = "Choose 1 option"
	default      = "option1"
	style        = "bulltes|dropdown"  # defaults to bullets to keep up the old behavior 
	option {
		name  = "foo"
		value = "option1"
	}
	option {
		name  = "bar"
		value = "option2"	
	}
	...
	option {
		name  = "bar"
		value = "option10"	
	}
}
```
cc: @timquinlan

**Comment by matifali** [comment]
@sreya, are we planning to do it automatically, or want to give the user an option to choose the style? Later needs to update the provider first. I personally prefer to handle in frontend depending on the number of `option` blocks.

**Comment by sreya** [comment]
Yeah that's what we're planning on doing. No provider changes. cc @aslilac

**Comment by aslilac** [comment]
I thought we decided to punt on this for now

**Comment by sreya** [comment]
Oh whoops you're right. We can maybe circle back around to this on the spillover week for the sprint.

**Comment by NiklasRosenstein** [comment]
Hi, I was about to file this as a feature request but I'll just bring this issue back up to attention instead. 😃 

I think it would be nice if the dropdown field would also allow entering text to filter choices. An option field in my template has 20+ options and it's not a great UX to parse that list.

<img width="645" alt="image" src="https://github.com/coder/coder/assets/1318438/432553d1-dee5-4151-aaee-813b1d76b817">

_(more below when scrolling)_

**Comment by matifali** [comment]
I agree. A drop-down list with a search bar to filter options as we type would be great UX.

**Comment by mattlqx** [comment]
I would also greatly appreciate this. I would also go so far as to suggest an optional typeahead input as we have many different VM image names that would be handy to be able to just start typing and the list would update to only those that include the substring.

**Comment by matifali** [comment]
We should add this if it's an easy lift or handle this case as part of #5540 and #11275

**Comment by matifali** [comment]
> I would also greatly appreciate this. I would also go so far as to suggest an optional typeahead input as we have many different VM image names that would be handy to be able to just start typing and the list would update to only those that include the substring.

This could be a separate request to allow search through coder_parameter options.

**Comment by stirby** [comment]
Completed with Dynamic Parameters!
